### ISSUE TITLE
graph partitioning method for data association

### ISSUE THREAD

**Issue by linye-boli** [longcw/crossview_3d_pose_tracking#9]
Hi, I found you refer to paper "Tracking Multiple People Online and in Real Time", which use "AL-ICM" method to solve correlation clustering problem, do you have the software which implements "AL-ICM" in python or other languages. I stuck on graph partition part

**Comment by B10532021** [comment]
Him I've wondered whether you solve correlation problem, stuck on this part either.
Thanks

**Comment by longcw** [comment]
Hi guys, "Tracking Multiple People Online and in Real Time" used both binary integer programming solver and "AL-ICM" for tracking. They used to have a matlab version of the source code. Here is my ported BIP solver in python https://gist.github.com/longcw/654a86ffe11122079040a7615c99a627

It requires cvxopt and GLPK, hope it's helpful for you!

**Comment by B10532021** [comment]
Hi @longcw,
Its definitely helpful! I'll take a look at it.
Thank you very much!

Best Regards

**Comment by fandulu** [comment]
Hey @longcw, I also have a question here. Could I ask why applying two kinds of matching, as Hungarian and graphpartition in the paper algorithm 1? Is the graphpartition just used for initialization like mvpose, then the following matching is Hungarian?

**Comment by longcw** [comment]
@fandulu Yes, in the paper the graph partition is only for target creation. Hungarian is enough for the bipartite matching problem, but if you would like to use some different matching method that considering multiple frames or multiple views, or one person is split into multiple parts, then you may need the graph partition.

**Comment by fandulu** [comment]
@longcw， thanks for your answer!
